ui/components/data-sources/DataSourcesList.js

```diff
@@ -1,11 +1,48 @@
 import React from "react";
 import { Query } from "react-apollo";
-import { ListView } from "patternfly-react";
+import {
+    ListView,
+    Button,
+    EmptyState,
+    EmptyStateIcon,
+    EmptyStateTitle,
+    EmptyStateInfo,
+    EmptyStateHelp,
+    EmptyStateAction
+} from "patternfly-react";
 
 import GetDataSources from "../../graphql/GetDataSources.graphql";
 import { DataSourcesListItem } from "./DataSourcesListItem";
 
-const DataSourcesList = ({ filter }) => (
+const EmptyList = ({ action }) => (
+    <EmptyState style={{ margin: "15px" }}>
+        <EmptyStateIcon />
+        <EmptyStateTitle>
+                No Data Sources defined
+        </EmptyStateTitle>
+        <EmptyStateInfo>
+                Data Sources are used to store and retrieve your data.
+                You should define at least one in order to use it in a resolver mapping.
+        </EmptyStateInfo>
+        <EmptyStateHelp>
+                For more information have a look at the
+            <a href="https://docs.aerogear.org">&nbsp;docs</a>
+        </EmptyStateHelp>
+        <EmptyStateAction>
+            <Button
+                bsStyle="primary"
+                bsSize="large"
+                onClick={() => {
```

Comment:
onClick={action}
But, since this component is closely related with "adding a data source", the callback should be called in accordance.